sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetQuerySuite.scala

```diff
@@ -804,6 +804,162 @@ abstract class ParquetQuerySuite extends QueryTest with ParquetTest with SharedS
     }
   }
 
+  test("SPARK-31116: Select simple parquet columns correctly in case insensitive manner") {
```

Comment:
Could you move new test cases into `FileBasedDataSourceSuite` and run with `Orc/Parquet/Json` at least?

Reply:
I'll test soon. however, could these new test cases apply to Orc and Json also?

Reply:
Yes. Please start with all of three and comment out if ORC/Json fails.

Reply:
I tested ORC and Json file format and there exist some failures.

## Json test failure
Json passed case sensitive cases, but it failed in case insensitive case
```
[info] - SPARK-31116: Select simple columns correctly in case insensitive manner *** FAILED *** (4 seconds, 277 milliseconds)
[info]   Results do not match for query:
[info]   Timezone: sun.util.calendar.ZoneInfo[id="America/Los_Angeles",offset=-28800000,dstSavings=3600000,useDaylight=true,transitions=185,lastRule=java.util.SimpleTimeZone[id=America/Los_Angeles,offset=-28800000,dstSavings=3600000,useDaylight=true,startYear=0,startMode=3,startMonth=2,startDay=8,startDayOfWeek=1,startTime=7200000,startTimeMode=0,endMode=3,endMonth=10,endDay=1,endDayOfWeek=1,endTime=7200000,endTimeMode=0]]
[info]   Timezone Env:
[info]
[info]   == Parsed Logical Plan ==
[info]   Relation[camelcase#56] json
[info]
[info]   == Analyzed Logical Plan ==
[info]   camelcase: string
[info]   Relation[camelcase#56] json
[info]
[info]   == Optimized Logical Plan ==
[info]   Relation[camelcase#56] json
[info]
[info]   == Physical Plan ==
[info]   FileScan json [camelcase#56] Batched: false, DataFilters: [], Format: JSON, Location: InMemoryFileIndex[file:/Users/kimtkyeom/Dev/spark_devel/target/tmp/spark-95f1357a-85c9-444f-bdcc-..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<camelcase:string>
[info]
[info]   == Results ==
[info]
[info]   == Results ==
[info]   !== Correct Answer - 1 ==   == Spark Answer - 1 ==
[info]   !struct<>                   struct<camelcase:string>
[info]   ![A]                        [null] (QueryTest.scala:248)
```

```
[info] - SPARK-31116: Select nested columns correctly in case insensitive manner *** FAILED *** (2 seconds, 117 milliseconds)
[info]   Results do not match for query:
[info]   Timezone: sun.util.calendar.ZoneInfo[id="America/Los_Angeles",offset=-28800000,dstSavings=3600000,useDaylight=true,transitions=185,lastRule=java.util.SimpleTimeZone[id=America/Los_Angeles,offset=-28800000,dstSavings=3600000,useDaylight=true,startYear=0,startMode=3,startMonth=2,startDay=8,startDayOfWeek=1,startTime=7200000,startTimeMode=0,endMode=3,endMonth=10,endDay=1,endDayOfWeek=1,endTime=7200000,endTimeMode=0]]
[info]   Timezone Env:
[info]
[info]   == Parsed Logical Plan ==
[info]   Relation[StructColumn#147] json
[info]
[info]   == Analyzed Logical Plan ==
[info]   StructColumn: struct<LowerCase:bigint,camelcase:bigint>
[info]   Relation[StructColumn#147] json
[info]
[info]   == Optimized Logical Plan ==
[info]   Relation[StructColumn#147] json
[info]
[info]   == Physical Plan ==
[info]   FileScan json [StructColumn#147] Batched: false, DataFilters: [], Format: JSON, Location: InMemoryFileIndex[file:/Users/kimtkyeom/Dev/spark_devel/target/tmp/spark-f9ecd1a4-e5aa-4dd7-bdfd-..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<StructColumn:struct<LowerCase:bigint,camelcase:bigint>>
[info]
[info]   == Results ==
[info]
[info]   == Results ==
[info]   !== Correct Answer - 1 ==   == Spark Answer - 1 ==
[info]   !struct<>                   struct<StructColumn:struct<LowerCase:bigint,camelcase:bigint>>
[info]   ![[0,1]]                    [[null,null]] (QueryTest.scala:248)
```
## ORC test failure
ORC passed case insensitive test cases, but it failed case insensitive manner.
```
[info] - SPARK-31116: Select nested columns correctly in case sensitive manner *** FAILED *** (871 milliseconds)
[info]   Results do not match for query:
[info]   Timezone: sun.util.calendar.ZoneInfo[id="America/Los_Angeles",offset=-28800000,dstSavings=3600000,useDaylight=true,transitions=185,lastRule=java.util.SimpleTimeZone[id=America/Los_Angeles,offset=-28800000,dstSavings=3600000,useDaylight=true,startYear=0,startMode=3,startMonth=2,startDay=8,startDayOfWeek=1,startTime=7200000,startTimeMode=0,endMode=3,endMonth=10,endDay=1,endDayOfWeek=1,endTime=7200000,endTimeMode=0]]
[info]   Timezone Env:
[info]
[info]   == Parsed Logical Plan ==
[info]   Relation[StructColumn#329] json
[info]
[info]   == Analyzed Logical Plan ==
[info]   StructColumn: struct<LowerCase:bigint,camelcase:bigint>
[info]   Relation[StructColumn#329] json
[info]
[info]   == Optimized Logical Plan ==
[info]   Relation[StructColumn#329] json
[info]
[info]   == Physical Plan ==
[info]   FileScan json [StructColumn#329] Batched: false, DataFilters: [], Format: JSON, Location: InMemoryFileIndex[file:/Users/kimtkyeom/Dev/spark_devel/target/tmp/spark-612baf76-a9d0-41e5-89f4-..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<StructColumn:struct<LowerCase:bigint,camelcase:bigint>>
[info]
[info]   == Results ==
[info]
[info]   == Results ==
[info]   !== Correct Answer - 1 ==   == Spark Answer - 1 ==
[info]   !struct<>                   struct<StructColumn:struct<LowerCase:bigint,camelcase:bigint>>
[info]   ![null]                     [[null,null]] (QueryTest.scala:248)
```
But i think ORC failure is due to difference between materializing Row. Is there clean way to test properly?

Reply:
In addition, I noticed that json does not follow case sensitivity even in spark 2.4.4. Below is my local machine test using spark-shell
```
20/03/12 19:20:19 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
20/03/12 19:20:24 WARN util.Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
Spark context Web UI available at http://61.75.36.130:4041
Spark context available as 'sc' (master = local[*], app id = local-1584008425035).
Spark session available as 'spark'.
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 2.4.4
      /_/

Using Scala version 2.11.12 (OpenJDK 64-Bit Server VM, Java 1.8.0_222)
Type in expressions to have them evaluated.
Type :help for more information.

scala> val df = Seq("A").toDF("camelCase")
df: org.apache.spark.sql.DataFrame = [camelCase: string]

scala> df.write.format("json").save("./json_simple")

scala> import org.apache.spark.sql.types._
import org.apache.spark.sql.types._

scala> val sch2 = new StructType().add("camelcase", StringType)
sch2: org.apache.spark.sql.types.StructType = StructType(StructField(camelcase,StringType,true))

scala> spark.read.format("json").schema(sch2).load("./json_simple").show()
+---------+
|camelcase|
+---------+
|     null.     |
+---------+
```

Reply:
Thank you for checking. Could you file a JIRA for regressions only?

Reply:
Could you update your PR?

Reply:
Ok, I updated my PR and create jira issue. As other file formats (orc and json) also fail these test cases I omit to check these formats by now, just moved current test cases into FileBasedDataSourceSuite. I think it would be added when regression will be fixed

Reply:
Was the ORC case only when `spark.sql.optimizer.nestedSchemaPruning.enabled` is enabled?

Reply:
I checked test, but it produce same result above regardless of nestedSchemaPruning option